### PR TITLE
[API] Promotion GET Collection

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
|-----------------|-----
| Branch?         | api-platform-3 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no 
| Related tickets | continuation of https://github.com/Sylius/Sylius/pull/16613, related to https://github.com/Sylius/Sylius/pull/16622, https://github.com/Sylius/Sylius/pull/16624
| License         | MIT

Current endpoints state:
image

### CODE REVIEW

**Review by mpysiak — APPROVED** [review]

File: disabledTests/Api/Admin/PromotionsTest.php
Comment (Wojdylak) [inline disabledTests/Api/Admin/PromotionsTest.php:1]: This is the last PR, so this file should be removed.

File: src/Sylius/Behat/Resources/config/services/contexts/api/admin.xml [hunk 1]
```diff
@@ -245,22 +245,21 @@
 <!--            <argument>%sylius.model.zone_member.class%</argument>-->
 <!--        </service>-->
 
-<!--        <service id="Sylius\Behat\Context\Api\Admin\RemovingProductContext">-->
-<!--            <argument type="service" id="sylius.behat.api_platform_client.admin" />-->
-<!--            <argument type="service" id="Sylius\Behat\Client\ResponseCheckerInterface" />-->
-<!--        </service>-->
+        <service id="Sylius\Behat\Context\Api\Admin\RemovingProductContext">
```
Comment (Wojdylak) [inline src/Sylius/Behat/Resources/config/services/contexts/api/admin.xml]:
```suggestion
        <service id="..." class="Sylius\Behat\Context\Api\Admin\RemovingProductContext">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 2]
```diff
@@ -314,19 +314,19 @@
 <!--            <tag name="api_platform.filter" />-->
 <!--        </service>-->
 
-<!--        <service id="sylius.api.promotion_coupon_search_filter" parent="api_platform.doctrine.orm.search_filter" public="true">-->
-<!--            <argument type="collection">-->
-<!--                <argument key="coupons.code">exact</argument>-->
-<!--            </argument>-->
-<!--            <tag name="api_platform.filter" />-->
-<!--        </service>-->
+        <service id="sylius_api.promotion_coupon_search_filter" parent="api_platform.doctrine.orm.search_filter" public="true">
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]:
```suggestion
        <service id="sylius_api.search_filter.promotion" parent="api_platform.doctrine.orm.search_filter" public="true">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 3]
```diff
@@ -314,19 +314,19 @@
...
+            <tag name="api_platform.filter" />
+        </service>
 
-<!--        <service id="sylius.api.promotion_order_filter" parent="api_platform.doctrine.orm.order_filter" public="true">-->
-<!--            <argument type="collection">-->
-<!--                <argument key="priority" />-->
-<!--            </argument>-->
-<!--            <tag name="api_platform.filter" />-->
-<!--        </service>-->
+        <service id="sylius_api.promotion_order_filter" parent="api_platform.doctrine.orm.order_filter" public="true">
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]:
```suggestion
        <service id="sylius_api.order_filter.promotion" parent="api_platform.doctrine.orm.order_filter" public="true">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 4]
```diff
@@ -408,7 +408,7 @@
             <tag name="api_platform.filter" />
         </service>
 
-        <service id="sylius.api.exists_filter.archived_at" parent="api_platform.doctrine.orm.exists_filter" public="true">
+        <service id="sylius_api.exists_filter.archived_at" parent="api_platform.doctrine.orm.exists_filter" public="true">
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]:
```suggestion
        <service id="sylius_api.exists_filter.promotion" parent="api_platform.doctrine.orm.exists_filter" public="true">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/extensions.xml [hunk 5]
```diff
@@ -135,10 +135,10 @@
             <tag name="api_platform.doctrine.orm.query_extension.item" />
         </service>
 
-<!--        <service id="sylius_api.doctrine.orm.query_extension.common.promotion.non_archived" class="Sylius\Bundle\ApiBundle\Doctrine\ORM\QueryExtension\Common\Promotion\NonArchivedExtension">-->
-<!--            <argument>%sylius.model.promotion.class%</argument>-->
-<!--            <tag name="api_platform.doctrine.orm.query_extension.collection" />-->
-<!--        </service>-->
+        <service id="sylius_api.doctrine.orm.query_extension.common.promotion.non_archived" class="Sylius\Bundle\ApiBundle\Doctrine\ORM\QueryExtension\Common\Promotion\NonArchivedExtension">
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/extensions.xml]: Would you like to handle the TODO task in this service?

File: disabledTests/Api/Admin/PromotionsTest.php
Comment (TheMilek) [reply disabledTests/Api/Admin/PromotionsTest.php:1]: Yeah i was planning to make separate PR for that, but yeah it is no longer that long to make it somewhere else 🤔

**Review by Wojdylak — APPROVED** [review]

**Pull request merged**